### PR TITLE
google-authenticator-libpam 1.05 (new formula)

### PR DESCRIPTION
**Pull request by jabenninghoff**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
macOS build of Google Authenticator example PAM module.

### CODE REVIEW

**Comment by jabenninghoff** [comment]
The included test just checks "--help"; which isn't a good test, but since google-authenticator writes a file to the user's home directory in normal use, I couldn't see an easy way to write a better test. I'd appreciate pointers for that. Thanks!

**Comment by ilovezfs** [comment]
>but since google-authenticator writes a file to the user's home directory in normal use

During `test do` the environment variable `HOME` is set equal to the `testpath` so it shouldn't be an issue unless it's ignoring `HOME`.

**Comment by jabenninghoff** [comment]
Thanks, I'll work on an updated test. I also need to add a fix to the formula as the code looks for version 3 of libqrencode specifically (current version is 4).

**Comment by ilovezfs** [comment]
Is version 4 support in master yet?

**Comment by jabenninghoff** [comment]
I've tested with v4 and it appears to work with no issues. I'm going to submit a PR to look for `libqrencode.dylib` instead of `libqrencode.3.dylib`.

File: Formula/google-authenticator-libpam.rb
Comment (DomT4) [inline Formula/google-authenticator-libpam.rb]: Don't need this empty line.

File: Formula/google-authenticator-libpam.rb
Comment (DomT4) [inline Formula/google-authenticator-libpam.rb]: You can't assume `/usr/local`. Use `#{opt_lib}`.

File: Formula/google-authenticator-libpam.rb
Comment (DomT4) [inline Formula/google-authenticator-libpam.rb]: Not sure any of the caveats below this are required to be honest, but I'll let ILZ weigh in.

**Comment by jabenninghoff** [comment]
OK I've updated the PR:
* added fix to support current version (4) of libqrencode in homebrew (upstream PR submitted)
* removed extra whitespace
* changed caveat to use `#{opt_lib}`
* better test

Let me know if I should remove the caveat altogether. Thanks!

**Comment by jabenninghoff** [comment]
OK,
I've removed the extra caveat as upon reflection, I agreed it was not necessary.
The upstream PR was accepted, and I've [requested](https://github.com/google/google-authenticator-libpam/issues/84) a new release. Upon release, I can update the version and remove the `inreplace`.

Is this OK to merge?

File: Formula/google-authenticator-libpam.rb
Comment (DomT4) [inline Formula/google-authenticator-libpam.rb]: Not a big thing but not really worth `cd`ing into a directory to execute an inreplace. Just do:
```ruby
inreplace "src/google-authenticator.c",
          "libqrencode.3.dylib",
          "libqrencode.4.dylib
```

**Comment by ilovezfs** [comment]
We'll need to wait for that new release.

**Comment by DomT4** [comment]
Cool. You can ignore my advice then @jabenninghoff 😄.

**Comment by jabenninghoff** [comment]
OK! I'll update the PR after the new release.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@stale why don't you learn how to read?

**Comment by jabenninghoff** [comment]
@ilovezfs thanks for the bump on both this and the upstream request.

**Comment by ilovezfs** [comment]
@jabenninghoff you're welcome! [1.05](https://github.com/google/google-authenticator-libpam/archive/1.05.tar.gz) is now out.

**Comment by jabenninghoff** [comment]
Hi this is updated to 1.05. Should be ready to go, not sure why it hasn't built...

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by jabenninghoff** [comment]
Hi, @ilovezfs all tests have passed, let me know if there's anything else needed to merge!

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @jabenninghoff! :confetti_ball:

**Pull request closed without merging**
